### ISSUE TITLE
SP-Initiated SSO using java-saml-toolkit does not redirect to attribute consumer service

### ISSUE THREAD

**Issue by nukalaharish** [onelogin/java-saml#202]
Hello Team,

We created an app as part of our company applications which I integrated with java-saml-toolkit latest version in the repo.

 App works flawless from OneLogin UI ( Idp- Initiated SSO ) and successfully redirects to the acs.jsp and then on to my redirect URL but throws an error when I try SP-Initiated SSO. 

After trying multiple times with different approach what I noticed is that the request is not making to the attribute consumer service which indeed redirects to my landing page. I have a personal dev account which I configured with same SP values and that works flawless with both Idp-Initiated SSO and SP-Initiated SSO.

I cross checked onelogin.saml.properties file multiple times to see if I missed anything from the instructions specified but everything is correct. I'm out of options. 

Please find the attached onelogin.saml.properties file, saml response, debug logs and screenshots.[
[saml_response.txt](https://github.com/onelogin/java-saml/files/2589915/saml_response.txt)

[debug.txt](https://github.com/onelogin/java-saml/files/2589925/debug.txt)
[onelogin.saml.properties.txt](https://github.com/onelogin/java-saml/files/2589927/onelogin.saml.properties.txt)
![sp config - nonprodbicgraphic](https://user-images.githubusercontent.com/11935914/48632141-262e5a80-e98e-11e8-8105-2b29e7bfd804.PNG)
![sp config - harish-dev onelogin](https://user-images.githubusercontent.com/11935914/48632143-2890b480-e98e-11e8-94df-a0ae82bfb549.PNG)

](url)

**Comment by pitbulk** [comment]
Hi, 

The Destination value of the SAMLResponse (http://devwcsedit.bicgraphic.com/cs/bic_jsp/acs.jsp) seems ok.

Can you provide the SAML Trace? Also, on the logs I don't see any SAML error, can you detail what happening and what system throw the error?

**Comment by nukalaharish** [comment]
I think I figured out the issue. I'm added as a user to a role which is assigned to the app and that is what is causing the problem. If a role is not assigned to the app and the user is added directly to the app, it seems to work. Do you have any idea how to handle this using toolkit code?

**Comment by pitbulk** [comment]
No idea how that is related to the SAML protocol/toolkit.
That seems something specific of the app.

**Comment by nukalaharish** [comment]
I too think it is not toolkit specific issue. I was just wondering if anyone encountered the same issue and figured out a workaround. 

I finally found a workaround, don't know if this is the right way to do. I modified the code in my application to redirect to OneLogin endpoint of my application (https://nonprodbicgraphic.onelogin.com/client/apps/select/228802599) which indeed gets authenticated from IdP not from SP.
